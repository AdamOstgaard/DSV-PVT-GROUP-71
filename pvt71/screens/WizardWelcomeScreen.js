import React from "react";
import { StyleSheet, Text, View, ScrollView } from "react-native";
import AppSingleButton from "../components/AppSingleButton";
import {Permissions} from "expo";

export default class WizardWelcomeScreen extends React.Component {
  constructor(props) {
    super(props);
    this.handlePress = this.handlePress.bind(this);
  }

  static navigationOptions = {
    header: null,
  };

  handlePress()  {
    Permissions.askAsync(Permissions.LOCATION),
    this.props.navigation.navigate("WizardVerifyContactScreen");
}

  render() {
    return (
      <View style={styles.container}>
      <ScrollView>
        <Text style={styles.heading}>Välkommen!</Text>

        <Text style={styles.infoText}>
          {""}
          Innan du kan börja använda applikationen behöver du sätta några grundläggande
          inställningar som nödkontakt, timerintervall och varningstimerintervall.
          Du behöver också ge applikationen godkännande för att kunna använda sig av din position via GPS.
          Ingen kommer att ha tillgång till din data förutom i eventuella nödsituationer då din position kan skickas till din nödkontakt.
        </Text>
        </ScrollView>
        <View style={styles.bottom}>
          <AppSingleButton
            title="Nästa"
            onPress={this.handlePress}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff"
  },
  heading: {
    fontSize: 35,
    //flex: 1,
    justifyContent: "flex-start",
    left: 10,
    marginTop: 50
  },
  infoText: {
    fontSize: 20,
    paddingLeft: 10,
    paddingTop: "5%",
    paddingBottom: "60%",
  },
  bottom: {
    flex: 1,
    justifyContent: "flex-end"
  }
});
