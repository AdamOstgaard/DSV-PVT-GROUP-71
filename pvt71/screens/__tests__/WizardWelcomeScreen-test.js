import "react-native";
import React from "react";
import WizardWelcomeScreen from "../WizardWelcomeScreen";
import { shallow } from "enzyme";

describe("Wizard WelcomeScreen tests", () => {
  describe("Wizard WelcomeScreen tests", () => {
    it("renders correctly", () => {
      const tree = shallow(<WizardWelcomeScreen />);
      expect(tree).toMatchSnapshot();
    });
  });

  describe("navgation test", () => {
    it("should navigate to contactwizard", () => {
      const navigation = {
        navigate: jest.fn()
      };

      const spy = jest.spyOn(navigation, "navigate");
      const wrapper = shallow(<WizardWelcomeScreen navigation={navigation} />);
      const appButton = wrapper.find("AppSingleButton");
      appButton.props().onPress();

      expect(spy).toBeCalledWith("WizardVerifyContactScreen");
      spy.mockRestore();
    });
  });
});
