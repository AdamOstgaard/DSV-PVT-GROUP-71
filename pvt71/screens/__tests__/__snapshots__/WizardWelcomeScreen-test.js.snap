// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wizard WelcomeScreen tests Wizard WelcomeScreen tests renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WizardWelcomeScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ScrollViewMock>
          <Component
            style={
              Object {
                "fontSize": 35,
                "justifyContent": "flex-start",
                "left": 10,
                "marginTop": 50,
              }
            }
          >
            Välkommen!
          </Component>
          <Component
            style={
              Object {
                "fontSize": 20,
                "paddingBottom": "60%",
                "paddingLeft": 10,
                "paddingTop": "5%",
              }
            }
          >
            
            Innan du kan börja använda applikationen behöver du sätta några grundläggande inställningar som nödkontakt, timerintervall och varningstimerintervall. Du behöver också ge applikationen godkännande för att kunna använda sig av din position via GPS. Ingen kommer att ha tillgång till din data förutom i eventuella nödsituationer då din position kan skickas till din nödkontakt.
          </Component>
        </ScrollViewMock>,
        <Component
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <AppSingleButton
            onPress={[Function]}
            title="Nästa"
          />
        </Component>,
      ],
      "style": Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "fontSize": 35,
                  "justifyContent": "flex-start",
                  "left": 10,
                  "marginTop": 50,
                }
              }
            >
              Välkommen!
            </Component>,
            <Component
              style={
                Object {
                  "fontSize": 20,
                  "paddingBottom": "60%",
                  "paddingLeft": 10,
                  "paddingTop": "5%",
                }
              }
            >
              
              Innan du kan börja använda applikationen behöver du sätta några grundläggande inställningar som nödkontakt, timerintervall och varningstimerintervall. Du behöver också ge applikationen godkännande för att kunna använda sig av din position via GPS. Ingen kommer att ha tillgång till din data förutom i eventuella nödsituationer då din position kan skickas till din nödkontakt.
            </Component>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Välkommen!",
              "style": Object {
                "fontSize": 35,
                "justifyContent": "flex-start",
                "left": 10,
                "marginTop": 50,
              },
            },
            "ref": null,
            "rendered": "Välkommen!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "",
                "Innan du kan börja använda applikationen behöver du sätta några grundläggande inställningar som nödkontakt, timerintervall och varningstimerintervall. Du behöver också ge applikationen godkännande för att kunna använda sig av din position via GPS. Ingen kommer att ha tillgång till din data förutom i eventuella nödsituationer då din position kan skickas till din nödkontakt.",
              ],
              "style": Object {
                "fontSize": 20,
                "paddingBottom": "60%",
                "paddingLeft": 10,
                "paddingTop": "5%",
              },
            },
            "ref": null,
            "rendered": Array [
              "",
              "Innan du kan börja använda applikationen behöver du sätta några grundläggande inställningar som nödkontakt, timerintervall och varningstimerintervall. Du behöver också ge applikationen godkännande för att kunna använda sig av din position via GPS. Ingen kommer att ha tillgång till din data förutom i eventuella nödsituationer då din position kan skickas till din nödkontakt.",
            ],
            "type": [Function],
          },
        ],
        "type": [MockFunction],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <AppSingleButton
            onPress={[Function]}
            title="Nästa"
          />,
          "style": Object {
            "flex": 1,
            "justifyContent": "flex-end",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
            "title": "Nästa",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ScrollViewMock>
            <Component
              style={
                Object {
                  "fontSize": 35,
                  "justifyContent": "flex-start",
                  "left": 10,
                  "marginTop": 50,
                }
              }
            >
              Välkommen!
            </Component>
            <Component
              style={
                Object {
                  "fontSize": 20,
                  "paddingBottom": "60%",
                  "paddingLeft": 10,
                  "paddingTop": "5%",
                }
              }
            >
              
              Innan du kan börja använda applikationen behöver du sätta några grundläggande inställningar som nödkontakt, timerintervall och varningstimerintervall. Du behöver också ge applikationen godkännande för att kunna använda sig av din position via GPS. Ingen kommer att ha tillgång till din data förutom i eventuella nödsituationer då din position kan skickas till din nödkontakt.
            </Component>
          </ScrollViewMock>,
          <Component
            style={
              Object {
                "flex": 1,
                "justifyContent": "flex-end",
              }
            }
          >
            <AppSingleButton
              onPress={[Function]}
              title="Nästa"
            />
          </Component>,
        ],
        "style": Object {
          "backgroundColor": "#fff",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Object {
                    "fontSize": 35,
                    "justifyContent": "flex-start",
                    "left": 10,
                    "marginTop": 50,
                  }
                }
              >
                Välkommen!
              </Component>,
              <Component
                style={
                  Object {
                    "fontSize": 20,
                    "paddingBottom": "60%",
                    "paddingLeft": 10,
                    "paddingTop": "5%",
                  }
                }
              >
                
                Innan du kan börja använda applikationen behöver du sätta några grundläggande inställningar som nödkontakt, timerintervall och varningstimerintervall. Du behöver också ge applikationen godkännande för att kunna använda sig av din position via GPS. Ingen kommer att ha tillgång till din data förutom i eventuella nödsituationer då din position kan skickas till din nödkontakt.
              </Component>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Välkommen!",
                "style": Object {
                  "fontSize": 35,
                  "justifyContent": "flex-start",
                  "left": 10,
                  "marginTop": 50,
                },
              },
              "ref": null,
              "rendered": "Välkommen!",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "",
                  "Innan du kan börja använda applikationen behöver du sätta några grundläggande inställningar som nödkontakt, timerintervall och varningstimerintervall. Du behöver också ge applikationen godkännande för att kunna använda sig av din position via GPS. Ingen kommer att ha tillgång till din data förutom i eventuella nödsituationer då din position kan skickas till din nödkontakt.",
                ],
                "style": Object {
                  "fontSize": 20,
                  "paddingBottom": "60%",
                  "paddingLeft": 10,
                  "paddingTop": "5%",
                },
              },
              "ref": null,
              "rendered": Array [
                "",
                "Innan du kan börja använda applikationen behöver du sätta några grundläggande inställningar som nödkontakt, timerintervall och varningstimerintervall. Du behöver också ge applikationen godkännande för att kunna använda sig av din position via GPS. Ingen kommer att ha tillgång till din data förutom i eventuella nödsituationer då din position kan skickas till din nödkontakt.",
              ],
              "type": [Function],
            },
          ],
          "type": [MockFunction],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <AppSingleButton
              onPress={[Function]}
              title="Nästa"
            />,
            "style": Object {
              "flex": 1,
              "justifyContent": "flex-end",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onPress": [Function],
              "title": "Nästa",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
